Mage.Sets/src/mage/cards/s/StingTheGlintingDagger.java

```diff
@@ -0,0 +1,118 @@
+package mage.cards.s;
+
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfCombatTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.decorator.ConditionalContinuousEffect;
+import mage.abilities.effects.common.UntapEnchantedEffect;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.abilities.keyword.FirstStrikeAbility;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.watchers.common.BlockingOrBlockedWatcher;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author Susucr
+ */
+public final class StingTheGlintingDagger extends CardImpl {
+
+    public StingTheGlintingDagger(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{2}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // Equipped creature gets +1/+1 and has haste.
+        Ability ability = new SimpleStaticAbility(new BoostEquippedEffect(1, 1));
+        ability.addEffect(new GainAbilityAttachedEffect(
+            HasteAbility.getInstance(), AttachmentType.EQUIPMENT
+        ).setText("and has haste"));
+        this.addAbility(ability);
+
+        // At the beginning of each combat, untap equipped creature.
+        this.addAbility(new BeginningOfCombatTriggeredAbility(
+            new UntapEnchantedEffect().setText("untap equipped creature"), //TODO: rename the effect and add AttachmentType to it?
```

Comment:
I agree that would be a sensible future refactor.

Reply:
I went ahead and did this in eeffafc

Reply:
thanks!
Was confused by the Predicate stuff and did not realize I did not went through with the small refactor, then went to sleep.